### ISSUE TITLE
App crashes when a dynamic feature module installation fails

### ISSUE THREAD

**Issue by negarq** [Sandor13/Translator#21]
Hi,

I've been checking this app, and when I try to choose the history screen, the app crashes and quits. Even restarting doesn't fix it, and I have to re-install the app to make it work again.
The reason seems to be that every click on the history screen triggers the installation of a dynamic feature module, and the installation request fails, which isn't handled properly in the code, as mentioned here:
https://developer.android.com/guide/playcore/play-feature-delivery#handle_request_errors

Can you verify/fix that?

**Comment by Sandor13** [comment]
Hey, thanks for your interest. I've run the app on both Emulator and real device and found zero issues. Can you please attach some stack trace for the crash you experiencing.

**Comment by negarq** [comment]
Thank you for your response.
When "startInstall" is called, if the installation request fails, for any reason e.g. network disconnection, the "addOnFailureListener" is called. The best way to handle it is to catch the different request error codes (e.g. -6 for network error) and handle it respectively. For instance, showing a dialog box notifying the user about the network failure and "check network connection" and "try again" options.
In the case of your app, when I requested to install the dynamic feature module, my wifi was disabled and the request failed, but instead of a proper notification, the app quit, and I had to reinstall it.
